pages/register.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import Head from 'next/head'
+import RegisterForm from '../components/RegisterForm'
+import styles from '../styles/Login.module.css'
+import router from 'next/router'
+
+const Register = () => { 
+  const registerRequest = (formData: React.FormEvent<HTMLInputElement>) => {
+    fetch("http://localhost:3001/api/signup", {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        user: {
+          username: formData.username,
+          password: formData.password,
+          email: formData.email,
+        },
+      }),
+    }).then((res) => {
+        localStorage.setItem("token", res.headers.get("Authorization"))
+        router.push("/discover")
+    }).catch((error) => {
+      throw new Error(error)

```

Comment:
Please add error handling. Iirc if you throw this error, your page will show an error screen.

Please tell the user that something in their form has an issue by:
- show the error inline in the form
- error alert box